src/micro_http/src/common/headers.rs

```diff
@@ -138,43 +143,76 @@ impl Headers {
                                 self.content_length = content_length;
                                 Ok(())
                             }
-                            Err(_) => Err(RequestError::InvalidHeader),
+                            Err(_) => Err(RequestError::HeaderError(
+                                HttpHeaderError::UnsupportedValue(
+                                    entry[0].to_string(),
+                                    entry[1].to_string(),
+                                ),
+                            )),
                         },
                         Header::ContentType => {
                             match MediaType::try_from(entry[1].trim().as_bytes()) {
                                 Ok(_) => Ok(()),
-                                Err(_) => Err(RequestError::UnsupportedHeader),
+                                Err(_) => {
+                                    Err(RequestError::HeaderError(HttpHeaderError::InvalidValue(
```

Comment:
Probably here we don't know if it's an invalid or an unsupported value. But we could go with unsupported to keep the previous behavior.